packages/x-data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -524,12 +524,13 @@ describe('<DataGrid /> - Pagination', () => {
     });
   });
 
-  it('should react to an update of rowCount', () => {
+  it('should react to an update of rowCount when `paginationMode = server`', () => {
```

Comment:
This is an updated behavior, now rowCount prop will be honoured when `paginationMode` is `server` since on client mode, internal calculation makes more sense for `rowCount` calculation as `rows` are available.
Any objections?